### PR TITLE
Feat: Add example for Mistral

### PR DESCRIPTION
**Pull request by NanoCode012**
Todo:
- [x] Update transformer commit

In future PR, allow sample_packing.

Upstream PR: https://github.com/huggingface/transformers/pull/26447

### CODE REVIEW

File: examples/mistral/config.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+base_model: mistralai/Mistral-7B-v0.1
+base_model_config: mistralai/Mistral-7B-v0.1
+model_type: MistralForCausalLM
```
Comment (winglian) [inline examples/mistral/config.yml:3]: Should be able to use AutoModelForCausalLM

**Comment by winglian** [comment]
let's use `[78dd120](https://github.com/huggingface/transformers/commit/78dd1202823ca035b9609ddbcdaac2945a6530ff)`

**Comment by winglian** [comment]
this PR should fix the issues we were having w transformers main that we had to pin it in the first place https://github.com/OpenAccess-AI-Collective/axolotl/pull/636

File: examples/mistral/config.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+strict: false
+
+datasets:
+  - path: mhenrichsen/alpaca_2k_test
+    type: alpaca
+dataset_prepared_path: last_run_prepared
+val_set_size: 0.01
+output_dir: ./out
+
+sequence_len: 4096
```
Comment (mhenrichsen) [inline examples/mistral/config.yml]: Isn't mistral 8k?

File: examples/mistral/config.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+sample_packing:
+pad_to_sequence_len:
+
+adapter:
+lora_model_dir:
+lora_r: 32
+lora_alpha: 16
+lora_dropout: 0.05
+lora_target_linear: true
+lora_fan_in_fan_out:
```
Comment (mhenrichsen) [inline examples/mistral/config.yml]: Maybe we should remove these if this is a full finetune example.

**Comment by winglian** [comment]
also, see #646

File: examples/mistral/config.yml (hunk 2 above)
Comment (NanoCode012) [reply examples/mistral/config.yml]: Thanks! As of that time, there was no flash, so I just thought to use a reasonable balanced number. However, I will now update it now that we have FA patch.

File: examples/mistral/config.yml (hunk 1 above)
Comment (NanoCode012) [reply examples/mistral/config.yml:3]: Yes, but I think it would be better to specify for later parsing (aka is_mistral_derived..)

**Pull request merged**